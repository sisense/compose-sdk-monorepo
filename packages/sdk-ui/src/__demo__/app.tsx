import { Alert, Tab, Tabs } from '@mui/material';
import { Box } from '@mui/system';
import { useState } from 'react';
import { SisenseContextProvider } from '../components/SisenseContextProvider';
import { SisenseContextProviderProps } from '../props';
import { ChartsFromExampleApp } from './pages/ChartsFromExampleApp';
import { ECommerceDemo } from './pages/ECommerceDemo';
import { MiscDemo } from './pages/MiscDemo';
import { WidgetDemo } from './pages/WidgetDemo';

// This page is meant to enable faster iterations during development than
// using react-ts-demo or other demo apps that require a built sdk-ui
// Simply add a page to the `pages` array and it will show up in `yarn dev`
// Suggest adding a router or at least sessionStorage var for selectedTabIndex
// if this becomes popular
const pages = [WidgetDemo, ECommerceDemo, ChartsFromExampleApp, MiscDemo];

const { VITE_APP_SISENSE_URL, VITE_APP_SISENSE_USERNAME, VITE_APP_SISENSE_PASSWORD } = import.meta
  .env;

export function App() {
  const [selectedTabIndex, setSelectedTabIndex] = useState(0);

  const shouldShowAlert =
    !VITE_APP_SISENSE_URL || !VITE_APP_SISENSE_USERNAME || !VITE_APP_SISENSE_PASSWORD;

  const sisenseContextProps: SisenseContextProviderProps = {
    url: VITE_APP_SISENSE_URL ?? '',
    username: VITE_APP_SISENSE_USERNAME,
    password: VITE_APP_SISENSE_PASSWORD,
    defaultDataSource: 'Sample ECommerce',
  };

  return (
    <>
      {shouldShowAlert && (
        <Alert severity="warning">
          One or more expected env vars are missing. You may want to check your
          <code>.env.local</code> file.
        </Alert>
      )}
      <SisenseContextProvider {...sisenseContextProps}>
        <div style={{ width: '100vw', height: '100vh' }}>
          <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
            <Tabs
              value={selectedTabIndex}
              onChange={(e, value: number) => setSelectedTabIndex(value)}
            >
              {pages.map((page, i) => (
                <Tab key={i} label={page.name} sx={{ textTransform: 'none' }} />
              ))}
            </Tabs>
          </Box>
          {pages.map((Page, i) => (
            <div key={i} hidden={selectedTabIndex !== i}>
              {selectedTabIndex === i && (
                <Box sx={{ p: 3 }}>
                  <Page />
                </Box>
              )}
            </div>
          ))}
        </div>
      </SisenseContextProvider>
    </>
  );
}
