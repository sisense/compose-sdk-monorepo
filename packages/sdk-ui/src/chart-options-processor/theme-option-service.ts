import merge from 'ts-deepmerge';
import { CompleteThemeSettings } from '../types';
import { HighchartsOptionsInternal } from './chart-options-service';
import { LegendSettings } from './translations/legend-section';

export const applyThemeToChart = (
  chartOptions: HighchartsOptionsInternal,
  themeSettings?: CompleteThemeSettings,
): HighchartsOptionsInternal => {
  if (!themeSettings) {
    return chartOptions;
  }

  const pie = {
    allowPointSelect: false,
    showInLegend: true,
    dataLabels: {
      style: {
        color: themeSettings.chart.textColor,
        fontFamily: themeSettings.typography.fontFamily,
      },
      showPercentLabels: true,
      showDecimals: false,
      pieMinimumFontSizeToTextLabel: 8,
    },
  };

  const chart = {
    backgroundColor: themeSettings.chart.backgroundColor,
  };

  const tooltip = {
    style: {
      fontFamily: themeSettings.typography.fontFamily,
    },
  };

  const legend = {
    itemStyle: {
      color: themeSettings.chart.textColor,
      fontFamily: themeSettings.typography.fontFamily,
    },
  } as LegendSettings;

  const seriesDataLabelsStyle = {
    style: {
      color: themeSettings.chart.textColor,
      fontFamily: themeSettings.typography.fontFamily,
    },
  };

  const plotOptions = {
    series: {
      dataLabels: seriesDataLabelsStyle,
    },
    pie,
  };

  const drilldown = {
    activeDataLabelStyle: {
      color: themeSettings.chart.textColor,
    },
    breadcrumbs: {
      buttonTheme: {
        fill: themeSettings.general.brandColor,
        stroke: themeSettings.general.brandColor,
        style: {
          color: themeSettings.general.primaryButtonTextColor,
        },
        states: {
          hover: {
            fill: themeSettings.general.primaryButtonHoverColor,
            stroke: themeSettings.general.primaryButtonHoverColor,
          },
        },
      },
    },
  };

  const basicOptions = {
    chart,
    tooltip,
    legend,
    plotOptions,
    drilldown,
  };

  const axisOptions = {
    labels: {
      style: {
        color: themeSettings.chart.textColor,
        fontFamily: themeSettings.typography.fontFamily,
      },
    },
    title: {
      style: {
        color: themeSettings.chart.textColor,
        fontFamily: themeSettings.typography.fontFamily,
      },
    },
  };
  const plotBandOptions = {
    color: themeSettings?.chart.backgroundColor,
    label: {
      style: {
        color: themeSettings.chart.textColor,
        fontFamily: themeSettings.typography.fontFamily,
      },
    },
  };

  const mergedOptions = merge(chartOptions, basicOptions);

  mergedOptions.xAxis = mergedOptions.xAxis?.map((axis) => {
    axis.plotBands = axis.plotBands?.map((plotBand) => {
      return merge(plotBand, plotBandOptions);
    });
    return merge(axis, axisOptions);
  });

  if (mergedOptions.yAxis) {
    mergedOptions.yAxis = mergedOptions.yAxis.map((axis) => {
      axis.plotBands = axis.plotBands?.map((plotBand) => {
        return merge(plotBand, plotBandOptions);
      });
      return merge(axis, axisOptions);
    });
  }

  return mergedOptions;
};
