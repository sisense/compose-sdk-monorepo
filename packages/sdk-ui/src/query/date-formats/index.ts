export type { DateFormat, DateConfig } from './apply-date-format';
export { applyDateFormat, defaultDateConfig } from './apply-date-format';
