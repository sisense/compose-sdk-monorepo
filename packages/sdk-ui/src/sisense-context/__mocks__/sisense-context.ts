export const useSisenseContext = vi.fn();
export const useSisenseContextMock = useSisenseContext;
