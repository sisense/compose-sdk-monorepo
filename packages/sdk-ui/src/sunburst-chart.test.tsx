import { render } from '@testing-library/react';
import { HighchartsOptions } from './chart-options-processor/chart-options-service';
import { useSisenseContext } from './sisense-context/sisense-context';
import { DAYS, JAN, MON } from './query/date-formats/apply-date-format';
import { getDefaultThemeSettings } from './chart-options-processor/theme-option-service';
import { getBaseDateFnsLocale } from './chart-data-processor/data-table-date-period';
import { ClientApplication } from './app/client-application';
import { SunburstChart } from './sunburst-chart';

vi.mock('./HighchartsWrapper', () => {
  return {
    HighchartsWrapper: ({ options }: { options: object }) => {
      return <div>{JSON.stringify(options)}</div>;
    },
  };
});

vi.mock('./sisense-context/sisense-context', async () => {
  const actual: typeof import('./sisense-context/sisense-context') = await vi.importActual(
    './sisense-context/sisense-context',
  );

  const useSisenseContextMock: typeof useSisenseContext = () => ({
    app: {
      settings: {
        dateConfig: {
          weekFirstDay: MON,
          isFiscalOn: false,
          fiscalMonth: JAN,
          selectedDateLevel: DAYS,
          timeZone: 'UTC',
        },
        serverThemeSettings: getDefaultThemeSettings(),
        locale: getBaseDateFnsLocale(),
      },
    } as ClientApplication,

    isInitialized: true,
    enableTracking: true,
  });

  return {
    ...actual,
    useSisenseContext: useSisenseContextMock,
  };
});

const dataSet = {
  columns: [
    { name: 'Gender', type: 'string' },
    { name: 'Years', type: 'date' },
    { name: 'Group', type: 'string' },
    { name: 'Quantity', type: 'number' },
    { name: 'Units', type: 'number' },
  ],
  rows: [
    ['Male', '2009', 'A', 6781, 10],
    ['Male', '2010', 'A', 4471, 70],
    ['Male', '2011', 'B', 1812, 50],
    ['Male', '2012', 'B', 5001, 60],
    ['Female', '2009', 'A', 1322, 10],
    ['Female', '2010', 'A', 2343, 70],
    ['Female', '2011', 'B', 1244, 50],
    ['Female', '2012', 'B', 3422, 60],
    ['Male', '2013', 'A', 2045, 40],
    ['Male', '2014', 'B', 3010, 90],
    ['Male', '2015', 'A', 5447, 80],
    ['Male', '2016', 'B', 4242, 70],
    ['Male', '2018', 'B', 936, 20],
  ],
};

const cat1 = {
  name: 'Years',
  type: 'date',
};

const cat2 = {
  name: 'Group',
  type: 'string',
};

const meas1 = {
  column: { name: 'Quantity', aggregation: 'sum' },
  showOnRightAxis: false,
};

describe('Sunburst Chart', () => {
  it('render a sunburst with single category', () => {
    render(
      <SunburstChart
        dataSet={dataSet}
        dataOptions={{ category: [cat1], value: [meas1] }}
        onBeforeRender={(options: HighchartsOptions) => {
          expect(options).toMatchSnapshot();
          return options;
        }}
      />,
    );
  });

  it('render a sunburst with two categories', () => {
    render(
      <SunburstChart
        dataSet={dataSet}
        dataOptions={{ category: [cat1, cat2], value: [meas1] }}
        onBeforeRender={(options: HighchartsOptions) => {
          expect(options).toMatchSnapshot();
          return options;
        }}
      />,
    );
  });

  it('render a sunburst with two categories and legend', () => {
    render(
      <SunburstChart
        dataSet={dataSet}
        dataOptions={{ category: [cat1, cat2], value: [meas1] }}
        styleOptions={{
          legend: {
            enabled: true,
            position: 'bottom',
          },
        }}
        onBeforeRender={(options: HighchartsOptions) => {
          expect(options).toMatchSnapshot();
          return options;
        }}
      />,
    );
  });
});
