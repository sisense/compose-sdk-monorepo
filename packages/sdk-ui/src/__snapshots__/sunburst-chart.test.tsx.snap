// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sunburst Chart > render a sunburst with single category 1`] = `
{
  "accessibility": {
    "enabled": false,
  },
  "boost": {
    "useGPUTranslations": true,
    "usePreAllocated": true,
  },
  "chart": {
    "alignTicks": false,
    "animation": {
      "duration": 300,
    },
    "backgroundColor": "#FFFFFF",
    "events": {},
    "polar": false,
    "spacing": [
      20,
      20,
      20,
      20,
    ],
    "type": "sunburst",
  },
  "drilldown": {
    "activeDataLabelStyle": {
      "color": "#000000",
    },
    "breadcrumbs": {
      "buttonTheme": {
        "fill": "#ffcb05",
        "states": {
          "hover": {
            "fill": "#f2b900",
            "stroke": "#f2b900",
          },
        },
        "stroke": "#ffcb05",
        "style": {
          "color": "#3a4356",
        },
      },
    },
  },
  "legend": {
    "align": "center",
    "enabled": true,
    "itemStyle": {
      "color": "#000000",
      "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
      "fontWeight": "normal",
    },
    "layout": "horizontal",
    "symbolRadius": 0,
    "verticalAlign": "bottom",
  },
  "plotOptions": {
    "pie": {
      "allowPointSelect": false,
      "dataLabels": {
        "pieMinimumFontSizeToTextLabel": 8,
        "showDecimals": false,
        "showPercentLabels": true,
        "style": {
          "color": "#000000",
          "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
        },
      },
      "showInLegend": true,
    },
    "series": {
      "animation": {
        "duration": 600,
      },
      "dataLabels": {
        "style": {
          "color": "#000000",
          "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
        },
      },
      "point": {
        "events": {},
      },
    },
    "sunburst": {
      "events": {
        "legendItemClick": [Function],
      },
    },
  },
  "series": [
    {
      "borderWidth": 0,
      "data": [
        {
          "custom": {
            "level": 0,
            "levelsCount": 1,
          },
          "id": "SUNBURST_ROOT_PARENT_ID",
          "name": "Quantity",
        },
        {
          "color": "#00cee6",
          "custom": {
            "level": 1,
            "levelsCount": 1,
          },
          "name": "2009",
          "parent": "SUNBURST_ROOT_PARENT_ID",
          "value": 8103,
        },
        {
          "color": "#00b9cf",
          "custom": {
            "level": 1,
            "levelsCount": 1,
          },
          "name": "2010",
          "parent": "SUNBURST_ROOT_PARENT_ID",
          "value": 6814,
        },
        {
          "color": "#00a5b8",
          "custom": {
            "level": 1,
            "levelsCount": 1,
          },
          "name": "2011",
          "parent": "SUNBURST_ROOT_PARENT_ID",
          "value": 3056,
        },
        {
          "color": "#0090a1",
          "custom": {
            "level": 1,
            "levelsCount": 1,
          },
          "name": "2012",
          "parent": "SUNBURST_ROOT_PARENT_ID",
          "value": 8423,
        },
        {
          "color": "#007c8a",
          "custom": {
            "level": 1,
            "levelsCount": 1,
          },
          "name": "2013",
          "parent": "SUNBURST_ROOT_PARENT_ID",
          "value": 2045,
        },
        {
          "color": "#006773",
          "custom": {
            "level": 1,
            "levelsCount": 1,
          },
          "name": "2014",
          "parent": "SUNBURST_ROOT_PARENT_ID",
          "value": 3010,
        },
        {
          "color": "#00525c",
          "custom": {
            "level": 1,
            "levelsCount": 1,
          },
          "name": "2015",
          "parent": "SUNBURST_ROOT_PARENT_ID",
          "value": 5447,
        },
        {
          "color": "#003e45",
          "custom": {
            "level": 1,
            "levelsCount": 1,
          },
          "name": "2016",
          "parent": "SUNBURST_ROOT_PARENT_ID",
          "value": 4242,
        },
        {
          "color": "#00292e",
          "custom": {
            "level": 1,
            "levelsCount": 1,
          },
          "name": "2018",
          "parent": "SUNBURST_ROOT_PARENT_ID",
          "value": 936,
        },
      ],
      "dataLabels": {
        "style": {
          "textOutline": "none",
        },
      },
      "levels": [
        {
          "color": "white",
          "dataLabels": {
            "className": "!csdk-overflow-visible",
            "enabled": true,
            "formatter": [Function],
            "useHTML": true,
          },
          "level": 1,
          "levelSize": {
            "unit": "pixels",
            "value": 60,
          },
        },
        {
          "dataLabels": {
            "enabled": false,
            "formatter": [Function],
          },
          "level": 2,
        },
      ],
      "name": "Root",
      "showInLegend": false,
      "type": "sunburst",
    },
    {
      "color": "#00cee6",
      "data": [],
      "name": "Years",
      "showInLegend": true,
      "states": {
        "hover": {
          "enabled": false,
        },
      },
    },
  ],
  "title": {
    "text": null,
  },
  "tooltip": {
    "animation": false,
    "backgroundColor": "#FFFFFF",
    "borderColor": "#CCCCCC",
    "borderRadius": 10,
    "borderWidth": 1,
    "formatter": [Function],
    "outside": true,
    "padding": 1,
    "style": {
      "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
    },
    "useHTML": true,
  },
  "xAxis": undefined,
}
`;

exports[`Sunburst Chart > render a sunburst with two categories 1`] = `
{
  "accessibility": {
    "enabled": false,
  },
  "boost": {
    "useGPUTranslations": true,
    "usePreAllocated": true,
  },
  "chart": {
    "alignTicks": false,
    "animation": {
      "duration": 300,
    },
    "backgroundColor": "#FFFFFF",
    "events": {},
    "polar": false,
    "spacing": [
      20,
      20,
      20,
      20,
    ],
    "type": "sunburst",
  },
  "drilldown": {
    "activeDataLabelStyle": {
      "color": "#000000",
    },
    "breadcrumbs": {
      "buttonTheme": {
        "fill": "#ffcb05",
        "states": {
          "hover": {
            "fill": "#f2b900",
            "stroke": "#f2b900",
          },
        },
        "stroke": "#ffcb05",
        "style": {
          "color": "#3a4356",
        },
      },
    },
  },
  "legend": {
    "align": "center",
    "enabled": true,
    "itemStyle": {
      "color": "#000000",
      "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
      "fontWeight": "normal",
    },
    "layout": "horizontal",
    "symbolRadius": 0,
    "verticalAlign": "bottom",
  },
  "plotOptions": {
    "pie": {
      "allowPointSelect": false,
      "dataLabels": {
        "pieMinimumFontSizeToTextLabel": 8,
        "showDecimals": false,
        "showPercentLabels": true,
        "style": {
          "color": "#000000",
          "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
        },
      },
      "showInLegend": true,
    },
    "series": {
      "animation": {
        "duration": 600,
      },
      "dataLabels": {
        "style": {
          "color": "#000000",
          "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
        },
      },
      "point": {
        "events": {},
      },
    },
    "sunburst": {
      "events": {
        "legendItemClick": [Function],
      },
    },
  },
  "series": [
    {
      "borderWidth": 0,
      "data": [
        {
          "custom": {
            "level": 0,
            "levelsCount": 2,
          },
          "id": "SUNBURST_ROOT_PARENT_ID",
          "name": "Quantity",
        },
        {
          "color": "#00cee6",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2009-01-01T00:00:00.000Z",
          "name": "2009",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#00b9cf",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2010-01-01T00:00:00.000Z",
          "name": "2010",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#00a5b8",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2011-01-01T00:00:00.000Z",
          "name": "2011",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#0090a1",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2012-01-01T00:00:00.000Z",
          "name": "2012",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#007c8a",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2013-01-01T00:00:00.000Z",
          "name": "2013",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#006773",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2014-01-01T00:00:00.000Z",
          "name": "2014",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#00525c",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2015-01-01T00:00:00.000Z",
          "name": "2015",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#003e45",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2016-01-01T00:00:00.000Z",
          "name": "2016",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#00292e",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2018-01-01T00:00:00.000Z",
          "name": "2018",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "A",
          "parent": "2009-01-01T00:00:00.000Z",
          "value": 8103,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "A",
          "parent": "2010-01-01T00:00:00.000Z",
          "value": 6814,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2011-01-01T00:00:00.000Z",
          "value": 3056,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2012-01-01T00:00:00.000Z",
          "value": 8423,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "A",
          "parent": "2013-01-01T00:00:00.000Z",
          "value": 2045,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2014-01-01T00:00:00.000Z",
          "value": 3010,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "A",
          "parent": "2015-01-01T00:00:00.000Z",
          "value": 5447,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2016-01-01T00:00:00.000Z",
          "value": 4242,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2018-01-01T00:00:00.000Z",
          "value": 936,
        },
      ],
      "dataLabels": {
        "style": {
          "textOutline": "none",
        },
      },
      "levels": [
        {
          "color": "white",
          "dataLabels": {
            "className": "!csdk-overflow-visible",
            "enabled": true,
            "formatter": [Function],
            "useHTML": true,
          },
          "level": 1,
          "levelSize": {
            "unit": "pixels",
            "value": 60,
          },
        },
        {
          "dataLabels": {
            "enabled": false,
            "formatter": [Function],
          },
          "level": 2,
        },
        {
          "dataLabels": {
            "enabled": false,
            "formatter": [Function],
          },
          "level": 3,
        },
      ],
      "name": "Root",
      "showInLegend": false,
      "type": "sunburst",
    },
    {
      "color": "#00cee6",
      "data": [],
      "name": "Years",
      "showInLegend": true,
      "states": {
        "hover": {
          "enabled": false,
        },
      },
    },
    {
      "color": "#9b9bd7",
      "data": [],
      "name": "Group",
      "showInLegend": true,
      "states": {
        "hover": {
          "enabled": false,
        },
      },
    },
  ],
  "title": {
    "text": null,
  },
  "tooltip": {
    "animation": false,
    "backgroundColor": "#FFFFFF",
    "borderColor": "#CCCCCC",
    "borderRadius": 10,
    "borderWidth": 1,
    "formatter": [Function],
    "outside": true,
    "padding": 1,
    "style": {
      "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
    },
    "useHTML": true,
  },
  "xAxis": undefined,
}
`;

exports[`Sunburst Chart > render a sunburst with two categories and legend 1`] = `
{
  "accessibility": {
    "enabled": false,
  },
  "boost": {
    "useGPUTranslations": true,
    "usePreAllocated": true,
  },
  "chart": {
    "alignTicks": false,
    "animation": {
      "duration": 300,
    },
    "backgroundColor": "#FFFFFF",
    "events": {},
    "polar": false,
    "spacing": [
      20,
      20,
      20,
      20,
    ],
    "type": "sunburst",
  },
  "drilldown": {
    "activeDataLabelStyle": {
      "color": "#000000",
    },
    "breadcrumbs": {
      "buttonTheme": {
        "fill": "#ffcb05",
        "states": {
          "hover": {
            "fill": "#f2b900",
            "stroke": "#f2b900",
          },
        },
        "stroke": "#ffcb05",
        "style": {
          "color": "#3a4356",
        },
      },
    },
  },
  "legend": {
    "align": "center",
    "enabled": true,
    "itemStyle": {
      "color": "#000000",
      "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
      "fontWeight": "normal",
    },
    "layout": "horizontal",
    "symbolRadius": 0,
    "verticalAlign": "bottom",
  },
  "plotOptions": {
    "pie": {
      "allowPointSelect": false,
      "dataLabels": {
        "pieMinimumFontSizeToTextLabel": 8,
        "showDecimals": false,
        "showPercentLabels": true,
        "style": {
          "color": "#000000",
          "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
        },
      },
      "showInLegend": true,
    },
    "series": {
      "animation": {
        "duration": 600,
      },
      "dataLabels": {
        "style": {
          "color": "#000000",
          "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
        },
      },
      "point": {
        "events": {},
      },
    },
    "sunburst": {
      "events": {
        "legendItemClick": [Function],
      },
    },
  },
  "series": [
    {
      "borderWidth": 0,
      "data": [
        {
          "custom": {
            "level": 0,
            "levelsCount": 2,
          },
          "id": "SUNBURST_ROOT_PARENT_ID",
          "name": "Quantity",
        },
        {
          "color": "#00cee6",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2009-01-01T00:00:00.000Z",
          "name": "2009",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#00b9cf",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2010-01-01T00:00:00.000Z",
          "name": "2010",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#00a5b8",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2011-01-01T00:00:00.000Z",
          "name": "2011",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#0090a1",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2012-01-01T00:00:00.000Z",
          "name": "2012",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#007c8a",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2013-01-01T00:00:00.000Z",
          "name": "2013",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#006773",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2014-01-01T00:00:00.000Z",
          "name": "2014",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#00525c",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2015-01-01T00:00:00.000Z",
          "name": "2015",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#003e45",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2016-01-01T00:00:00.000Z",
          "name": "2016",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#00292e",
          "custom": {
            "level": 1,
            "levelsCount": 2,
          },
          "id": "2018-01-01T00:00:00.000Z",
          "name": "2018",
          "parent": "SUNBURST_ROOT_PARENT_ID",
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "A",
          "parent": "2009-01-01T00:00:00.000Z",
          "value": 8103,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "A",
          "parent": "2010-01-01T00:00:00.000Z",
          "value": 6814,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2011-01-01T00:00:00.000Z",
          "value": 3056,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2012-01-01T00:00:00.000Z",
          "value": 8423,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "A",
          "parent": "2013-01-01T00:00:00.000Z",
          "value": 2045,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2014-01-01T00:00:00.000Z",
          "value": 3010,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "A",
          "parent": "2015-01-01T00:00:00.000Z",
          "value": 5447,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2016-01-01T00:00:00.000Z",
          "value": 4242,
        },
        {
          "color": "#9b9bd7",
          "custom": {
            "level": 2,
            "levelsCount": 2,
          },
          "name": "B",
          "parent": "2018-01-01T00:00:00.000Z",
          "value": 936,
        },
      ],
      "dataLabels": {
        "style": {
          "textOutline": "none",
        },
      },
      "levels": [
        {
          "color": "white",
          "dataLabels": {
            "className": "!csdk-overflow-visible",
            "enabled": true,
            "formatter": [Function],
            "useHTML": true,
          },
          "level": 1,
          "levelSize": {
            "unit": "pixels",
            "value": 60,
          },
        },
        {
          "dataLabels": {
            "enabled": false,
            "formatter": [Function],
          },
          "level": 2,
        },
        {
          "dataLabels": {
            "enabled": false,
            "formatter": [Function],
          },
          "level": 3,
        },
      ],
      "name": "Root",
      "showInLegend": false,
      "type": "sunburst",
    },
    {
      "color": "#00cee6",
      "data": [],
      "name": "Years",
      "showInLegend": true,
      "states": {
        "hover": {
          "enabled": false,
        },
      },
    },
    {
      "color": "#9b9bd7",
      "data": [],
      "name": "Group",
      "showInLegend": true,
      "states": {
        "hover": {
          "enabled": false,
        },
      },
    },
  ],
  "title": {
    "text": null,
  },
  "tooltip": {
    "animation": false,
    "backgroundColor": "#FFFFFF",
    "borderColor": "#CCCCCC",
    "borderRadius": 10,
    "borderWidth": 1,
    "formatter": [Function],
    "outside": true,
    "padding": 1,
    "style": {
      "fontFamily": "\\"Open Sans\\",\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif",
    },
    "useHTML": true,
  },
  "xAxis": undefined,
}
`;
